File: Erlang/bkk_crypt.erl
%%% BKK secure cipher module %

%% Usage, from erl...
%%
%% - compile:
%%
%%     > c(bkk_crypt).
%%
%% - run:
%%
%%     > bkk_crypt:encrypt("Haxxor protec, but he also attac").
%%
%% - to run tests:
%%
%%     > eunit:test(bkk_crypt).
%%
-module(bkk_crypt).

%% API
-export([encrypt/1]).

%% Tests
-include_lib("eunit/include/eunit.hrl").

encrypt(Message) ->
    Message.

-ifdef(EUNIT).
encrypt_test() ->
    ?assertEqual("password", bkk_crypt:encrypt("password")),
    ?assertNotEqual("7KVcbLRkKU15XiCRlTGuj0raudw+pl+SaGVnm456LoE=", bkk_crypt:encrypt("password")).
-endif.

File: Ruby/bkk_crypt.rb
###########################################################
# Ruby implementation of the famous hardened BKK cipher.
###########################################################
# Example:
# puts BKKCrypt::bkk_crypt("Safe and secure")
# puts "Haxxor protec, but he also attac".bkk_crypt
###########################################################
module BKKCrypt
    def bkk_crypt(*args)
        return self if args.empty?
        return args
    end
end

# Advanced BKK cryptos will make every object safe
class Object
    prepend BKKCrypt
end
